src/fastga_he/powertrain_builder/unit_tests/test_power_train_file.py

```diff
@@ -166,6 +178,168 @@ def test_power_train_file_connections():
         print("[" + om_output + ", " + om_input + "]")
 
 
+def test_power_train_file_connection_check_cache():
+    sample_power_train_file_path = pth.join(pth.dirname(__file__), "data", YML_FILE)
+    power_train_configurator = FASTGAHEPowerTrainConfigurator(
+        power_train_file_path=sample_power_train_file_path
+    )
+
+    power_train_configurator._get_components()
+    power_train_configurator._get_connections()
+
+    old_connection_check_cache = power_train_configurator._connection_check_cache
+
+    power_train_configurator._get_connections()
```

Comment:
Are you trying to check that the _cache mechnaism is correct in the sense that it doesn retrigger a check of the components ? If yes, I'd suggest testing the `_check_existing_instance`. Here you could trigger the recheck without knowing why